### PR TITLE
cleanup: Remove duplicate library names from `Cargo.toml`s

### PR DESCRIPTION
**Pull request by petrochenkov**

### CODE REVIEW

**Comment by matklad** [comment]
Should we do this after https://github.com/rust-lang/rust/pull/73265 perhaps? We'll be able to remove `path=` as well.

**Comment by tshepang** [comment]
@matklad is it a new feature for it to work without needing to specify `path=`?

**Comment by matklad** [comment]
No,  #73265 in its current form just proposes to leverage standard cargo's convetion, which doesn't require explicit config.

**Comment by petrochenkov** [comment]
>Should we do this after https://github.com/rust-lang/rust/pull/73265 perhaps?

Yeah, this can be done as a part of the std move (#73265) or as a part of the compiler move (majority of the changes), or separately.
Neither makes git history harder to follow, so there's probably not much difference.

**Comment by matklad** [comment]
There's differnce in how annoying is #73265 to rebase though :)

**Comment by nikomatsakis** [comment]
ok, marking as blocked on https://github.com/rust-lang/rust/pull/73265 I guess

**Comment by petrochenkov** [comment]
Blocked on https://github.com/rust-lang/rust/pull/74862.

File: compiler/rustc/Cargo.toml [hunk 1]
```diff
@@ -4,10 +4,6 @@ name = "rustc-main"
 version = "0.0.0"
 edition = '2018'
 
-[[bin]]
-name = "rustc_binary"
-path = "src/main.rs"
```
Comment (mark-i-m) [inline compiler/rustc/Cargo.toml:9]: Bootstrap actually assumes the name of the binary is `rustc_binary`, so you also need to change the package name on line 3 to `rustc_binary`, I think.

**Review by mark-i-m — CHANGES_REQUESTED** [review]

**Comment by mark-i-m** [comment]
Other than the above, r=me (but I don't have bors priveleges).

**Review by mark-i-m — APPROVED** [review]

**Comment by petrochenkov** [comment]
```
---- build::close_output stdout ----
thread 'build::close_output' panicked at 'lines differ:
   Compiling foo v0.1.0 (D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\cit\t215\foo)
error: The pipe is being closed. (os error 232)
warning: build failed, waiting for other jobs to finish...
hello stderr!
error: build failed
', src\tools\cargo\tests\testsuite\build.rs:5043:5
```
Looks spurious.
@bors retry

**Pull request merged**